chamge/amqp-edge-backend.c

```diff
@@ -268,11 +216,20 @@ _amqp_rpc_request (amqp_connection_state_t amqp_conn, guint channel,
 
   g_debug ("declare a queue for reply : %s", (gchar *) amqp_reply_queue.bytes);
 
-  /* send the message */
-  if (_amqp_property_set (&amqp_props, channel,
-          (gchar *) amqp_reply_queue.bytes) != CHAMGE_RETURN_OK) {
-    g_error ("create property failed.");
-    return NULL;
+  /* property setting to send rpc request */
+  {
+    memset (&amqp_props, 0, sizeof (amqp_basic_properties_t));
+    amqp_props._flags =
+        AMQP_BASIC_CONTENT_TYPE_FLAG | AMQP_BASIC_DELIVERY_MODE_FLAG;
+
+    amqp_props.content_type = amqp_cstring_bytes (DEFAULT_CONTENT_TYPE);
+    amqp_props.delivery_mode = 2;       /* persistent delivery mode */
+
+    amqp_props._flags |= AMQP_BASIC_REPLY_TO_FLAG;
+    amqp_props.reply_to = amqp_reply_queue;
+
+    amqp_props._flags |= AMQP_BASIC_CORRELATION_ID_FLAG;
+    amqp_props.correlation_id = amqp_cstring_bytes ("1");
```

Comment:
Yes, but it is temporary. HEAD was using channel for correlation_id. I was going to change the type of correlation_id from int to string and use edge_id for it. Correlation id is used for verifying that a response is matched for a request. If you have better idea for it, please let me know.

Reply:
If it is a unique id match message pair, I guess it would be better to randomize.
How about `g_uuid_string_random()` - https://developer.gnome.org/glib/stable/glib-GUuid.html#g-uuid-string-random

But because we are going to use hash, `HASH(uuid())` will be good in the manner of consistency.

Reply:
Ok, I will keep in mind for next patch.

Reply:
👍